src/containers/AccountConnection.jsx

```diff
@@ -102,7 +103,7 @@ class AccountConnection extends Component {
 
     return this.store.updateAccount(connector, account, values)
     .then(() => this.store.runAccount(connector, account))
-    .then(() => this.handleUpdateSuccess(account))
+    .then((job) => this.handleUpdateSuccess(account, job))
```

Comment:
I think that the `runAccount` store function should handle the job here.
If queued and timed out,`runAccount ` could `Promise.reject`, you catch a specific error status `timedOut` and continue to the `handleTimeout` here. In this way, this component will never know about a job here.
Otherwise `runAccount ` can `Promise.resolve` a boolean about timedOut and check for it in `handleUpdateSucess`.